File: packages/evals/tasks/simple_google_search.ts
import { EvalFunction } from "../types/evals";

export const simple_google_search: EvalFunction = async ({
  debugUrl,
  sessionUrl,
  v3,
  logger,
}) => {
  try {
    const page = v3.context.pages()[0];
    await page.goto(
      "https://browserbase.github.io/stagehand-eval-sites/sites/google/",
    );

    await v3.act('type "OpenAI" into the search bar');

    await v3.act("press enter");

    const expectedUrl =
      "https://browserbase.github.io/stagehand-eval-sites/sites/google/openai.html";
    const currentUrl = await page.url();

    return {
      _success: currentUrl.startsWith(expectedUrl),
      currentUrl,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } catch (error) {
    return {
      _success: false,
      error: error,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } finally {
    await v3.close();
  }
};

File: packages/sdk-ruby/evals/tasks/act/simple_google_search.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/simple_google_search.ts

Evals.define_task("simple_google_search") do |t|
  t.page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/google/")

  t.stagehand.act('type "OpenAI" into the search bar')

  t.stagehand.act("press enter")
  t.page.wait_for_timeout(3000)

  expected_url = "https://browserbase.github.io/stagehand-eval-sites/sites/google/openai.html"
  current_url = t.page.url

  { _success: current_url.start_with?(expected_url), currentUrl: current_url }
end
